wagtail/core/models.py

```diff
@@ -2738,11 +2740,60 @@ def cancel(self, user=None, resume=False):
         self.finished_at = timezone.now()
         self.save()
         if resume:
-            self.task.specific.start(self.workflow_state, user=user)
-        self.workflow_state.update(user=user)
+            self.workflow_state.update(user=user, next_task=self.task.specific)
+        else:
+            self.workflow_state.update(user=user)
         task_cancelled.send(sender=self.specific.__class__, instance=self.specific, user=user)
         return self
 
+    def copy(self, update_attrs={}, exclude_fields=None):
```

Comment:
I see what you mean - yes, I can do that. The reason I did this initially was to avoid the ClusterableModel-specific stuff (which is why I haven't included ParentalKeys - TaskState is not a ClusterableModel), but I imagine I could refactor the shared stuff out without much trouble